packages/fxa-settings/src/models/hooks.ts

```diff
@@ -82,3 +82,25 @@ export function useFtlMsgResolver() {
   const { l10n } = useLocalization();
   return new FtlMsgResolver(l10n, config.l10n.strict);
 }
+
+/**
+ * Hook to run a function on an interval.
+ * @param callback - function to call
+ * @param delay - interval in Ms to run, null to stop poll
+ */
+export function useInterval(callback: () => void, delay: number | null) {
+  const savedCallback = useRef(callback);
+
+  useEffect(() => {
+    savedCallback.current = callback;
+  }, [callback]);
+
+  useEffect(() => {
+    if (!delay && delay !== 0) {
+      return;
+    }
+
+    const id = window.setInterval(() => savedCallback.current(), delay);
```

Comment:
Since savedCallback is created with a useRef, do you have to make user that savedCallback.current hasn't been destroyed?

Reply:
Don't think it will hurt to add the check.